src/main/java/com/aws/greengrass/lifecyclemanager/GenericExternalService.java

```diff
@@ -120,9 +120,9 @@ protected GenericExternalService(Topics c, Topics privateSpace, Platform platfor
                 updateSystemResourceLimits();
             }
 
-            // Reinstall for changes to the install script or if the package version changed, or posixUser has changed
+            // Reinstall for changes to the install script or if the package version changed, or runWith user changed
             if (child.childOf(Lifecycle.LIFECYCLE_INSTALL_NAMESPACE_TOPIC) || child.childOf(VERSION_CONFIG_KEY)
-                    || child.childOf(POSIX_USER_KEY)) {
+                    || child.childOf(RUN_WITH_NAMESPACE_TOPIC) && !child.childOf(SYSTEM_RESOURCE_LIMITS_TOPICS)) {
```

Comment:
I hope this is correct, I would add clarifying parenthesis as long as pmd doesn't complain

Reply:
It did complain. I added suppression.